src/DefaultBuilder/src/StaticWebAssets/StaticWebAssetsHelper.cs

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Xml.Linq;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.FileProviders;
+
+namespace Microsoft.AspNetCore
+{
+    internal static class StaticWebAssetsHelper
+    {
+        private const string StaticWebAssetsManifestName = "Microsoft.AspNetCore.StaticWebAssets.xml";
+        private const string ManifestRootElementName = "StaticWebAssets";
+        private const string VersionAttributeName = "Version";
+        private const string ContentRootElementName = "ContentRoot";
+
+        internal static void UseStaticWebAssets(this IWebHostEnvironment environment)
+        {
+            using (var manifest = ResolveManifest(environment))
+            {
+                if (manifest != null)
+                {
+                    UseStaticWebAssetsCore(environment, manifest);
+                }
+            }
+        }
+
+        internal static void UseStaticWebAssetsCore(this IWebHostEnvironment environment, Stream manifest)
+        {
+            var staticWebAssetsFileProvider = new List<IFileProvider>();
+            var webRootFileProvider = environment.WebRootFileProvider;
+
+            var staticWebAssetContentRoots = Parse(manifest)
+                .Select(cr => new StaticWebAssetsFileProvider(cr.BasePath, cr.Path))
+                .OfType<IFileProvider>() // Upcast so we can insert on the resulting list.
+                .ToList();
+
+            if (staticWebAssetContentRoots.Count == 0)
+            {
+                return;
+            }
+            else
+            {
+                staticWebAssetContentRoots.Insert(0, webRootFileProvider);
+                environment.WebRootFileProvider = new CompositeFileProvider(staticWebAssetContentRoots);
+            }
+        }
+
+        internal static Stream ResolveManifest(IWebHostEnvironment environment)
+        {
+            var assembly = Assembly.Load(environment.ApplicationName);
+            if (assembly.GetManifestResourceNames().Any(a => a == StaticWebAssetsManifestName))
+            {
+                return assembly.GetManifestResourceStream(StaticWebAssetsManifestName);
```

Comment:
Do we need this check? I think `GetManifestResourceStream` would just do the right thing.

Reply:
We do. I think it returns a bogus/empty stream. My tests were failing without it.